### ISSUE TITLE
Notice: Undefined index: clean_url in system_urls_settings_submit() line 2151 of system.admin.inc

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#931]
After running the clear URLs test:

> Notice: Undefined index: clean_url in system_urls_settings_submit() (line 2151 of C:\WPN-XM\www\backdrop\core\modules\system\system.admin.inc).

**Comment by quicksketch** [comment]
Thanks @klonos. Really this whole form needs work. On some systems the clean URL check is entirely hidden, so I hadn't noticed the odd display of the form previously. Now we have "Use HTTPS for canonical URLs" showing but the button just says "Run the Clean URLs test", that doesn't make any sense.

We need to not only fix the submit handler to prevent the notice, but update the form so that the save button is always present, and the Run Clean URL Check button appears in addition only when needed.

**Comment by BWPanda** [comment]
Is work still needed here...? This is what I see if I manually set `$available` and `$conflict` to `FALSE`:

![image](https://user-images.githubusercontent.com/2385329/104113219-7a091400-534b-11eb-957f-3b13744abfb8.png)

**Comment by klonos** [comment]
When I filed this issue back in 2015, I was using Windows + https://wpn-xm.org. In that setup, clean URLs were not enabled by default, and when running the test, I'd get the error I've reported in the issue summary. I obviously have no easy way to reproduce this, and I'm not willing to spend the time setting up a virtual box to test this.

I think it's reasonable to close this, since nobody else reported this issue in 5 years now. I'm sure that if it comes up again, we'll reopen this issue. (I doubt that though, since the `system_urls_settings()` function specifies `$form['#config'] = 'system.core';` and `system_urls_settings_submit()` doesn't even exist in our codebase any more).

Having said that, there are certain things that we should fix - not as part of this issue here though; just listing them here as notes:

1. Seeing @quicksketch's comment:

   > ...update the form so that the save button is always present, and the Run Clean URL Check button appears in addition only when needed.

   This is now fixed, but the "Save configuration" button is shown as a secondary button, and the "Run the clean URL test" button as the primary submit button. We already have a UI pattern for similar forms, which we should match here. See `admin/config/development/performance`:

   ![image](https://user-images.githubusercontent.com/2423362/104128367-dd0ca200-536f-11eb-9bdd-3d61558b903d.png)

   ...or `admin/config/system/cron`:

   ![image](https://user-images.githubusercontent.com/2423362/104128438-2826b500-5370-11eb-9938-6ff09284bc0c.png)

1. Clicking the "Run test" button multiple times duplicates the warning and info massages.

1. The warning and help text for the clean URLs uses a hard-coded `example.com` domain. We should be using the actual domain/port, same as we do for the field prefixes in `admin/config/system/site-information`.

1. The help text for the canonical URLs needs some love:
   - The first bit (`This option makes Backdrop use HTTPS protocol for generated canonical URLs.`) provides no value, as it basically repeats the checkbox label (`Use HTTPS for canonical URLs`).
   - We could be pointing to online documentation, to explain what "canonical URLs" are.
   - The help text mentions `the variable https should be set to TRUE in your file settings.php`. We don't have variables, in Backdrop, so that should change to say that `` needs to be added to the settings.php file.
